### DISCUSSION TITLE
config_reader/writer without repo?

### DISCUSSION THREAD

**Discussion by jalanh11640** [gitpython-developers/GitPython#1573] [Q&A]
Is there a way to get the config_reader() and config_writer() without a repo?  I'm writing a tool to help users setup their .gitconfig file for our development environment.  They're doing this before cloning an repositories.  So I have no directory I can use for git.Repo('dir').

Thanks

**Comment by Byron** [comment]
It's close to impossible to find anything in the generated API docs, and even though I'd think it should be possible, it's hard to see without looking at the code itself.

In theory, a clone is `init + add remote + pull`, ignoring some special bits that might not be relevant for you. Thus it's possible to have a freshly initialized repo, adjust the configuration and then finish the clone. If that's not possible, I recommend checking the `config` module for code and see how the reader and writer can be initialised - getting a writer for a single file should be straightforward.

I hope that helps.
